mongo/integration/unified/session_operation_execution.go

```diff
@@ -103,7 +103,7 @@ func executeWithTransaction(ctx context.Context, op *operation, loopDone <-chan
 		return fmt.Errorf("error unmarshalling arguments to transactionOptions: %v", err)
 	}
 
-	_, err = sess.WithTransaction(ctx, func(sessCtx mongo.SessionContext) (interface{}, error) {
+	_, err = sess.WithTransaction(ctx, func(mongo.SessionContext) (interface{}, error) {
```

Comment:
```suggestion
	_, err = sess.WithTransaction(ctx, func(ctx mongo.SessionContext) (interface{}, error) {
```

I expect that all operations in the "withTransaction" need to use the context argument to run in the same transaction.

The "withTransaction" unified test operation may not be sufficiently exercised to make this behavior cause a test failure. But I suggest updating to prevent a difficult-to-diagnose error in the future.

Reply:
Good catch, that could definitely have led to a similar error.